src/webview/components/options-widget.tsx

```diff
@@ -116,11 +129,48 @@ export class OptionsWidget extends React.Component<OptionsWidgetProps, {}> {
         return errors;
     };
 
+    componentDidUpdate(prevProps: Readonly<OptionsWidgetProps>, prevState: Readonly<OptionsWidgetState>): void {
+        if (this.props.initialTitle !== prevProps.initialTitle) {
+            this.setState({ title: this.props.initialTitle });
+        }
+        if (!prevState.isTitleEditing && this.state.isTitleEditing) {
+            this.labelEditInput.current?.focus();
+            this.labelEditInput.current?.select();
+        }
+    }
+
     override render(): React.ReactNode {
         this.formConfig.initialValues = this.optionsFormValues;
+        const isLabelEditing = this.state.isTitleEditing;
 
         return (
             <div className='memory-options-widget px-4'>
+                <div className='title-container'>
+                    <InputText
+                        ref={this.labelEditInput}
+                        type="text"
```

Comment:
minor lint: we should probably stick to single quotes

Reply:
Nice catch! I'll fix this. We should probably add this as eslint rule.